browser/components/newtab/MozContentSearch.js

```diff
@@ -239,6 +243,21 @@ MozContentSearch.prototype = {
     }.bind(this));
   },
 
+  _fireStateChangeEvent(engines) {
+    Task.spawn(function* () {
+      const allEngines = engines.map(
+        engineDetails => this._storeEngine(engineDetails)
+      );
+      const safeArray = new this._win.Array();
+      safeArray.push(...allEngines);
+      const eventInit = {
+        engines: safeArray
+      };
+      const event = new this._win.MozVisibleSearchEnginesChangeEvent("visibleenginechange", eventInit);
```

Comment:
We also need to add an "onvisibleengineschange" EventHandler attribute on the IDL interface on which this is exposed. I.e.,

```
interface MozContentSearch {
  attribute EventHandler onvisiblesenginechange;
}
```